app/controllers/oai_controller.rb

```diff
@@ -170,8 +184,32 @@ def build_identify_response(earliest_date)
       end
     end.to_xml
   end
-  # rubocop:enable Metrics/AbcSize
-  # rubocop:enable Metrics/MethodLength
+
+  # See http://www.openarchives.org/OAI/openarchivesprotocol.html#ListMetadataFormats
+  def build_list_metadata_formats_response
+    Nokogiri::XML::Builder.new do |xml|
+      build_oai_response xml, list_metadata_formats_params do
+        xml.ListMetadataFormats do
+          xml.metadataFormat do
+            xml.metadataPrefix 'marc21'
+            xml.schema 'http://www.loc.gov/standards/marcxml/schema/MARC21slim.xsd'
+            xml.metadataNamespace 'http://www.loc.gov/MARC21/slim'
```

Comment:
i can't understand the difference between marc21 and marcxml, or whether this is right...

Reply:
I think what you have is correct. The terminology is confusing, but MARC21slim is an XML expression of MARC21.